server/src/main/java/io/pravega/schemaregistry/service/SchemaRegistryService.java

```diff
@@ -267,22 +269,25 @@ public SchemaRegistryService(SchemaStore store, ScheduledExecutorService executo
         // 2. get checker for serialization format.
         // validate schema against group compatibility policy on schema
         // 3. conditionally update the schema
-        return RETRY.runAsync(() -> store.getGroupEtag(namespace, group)
-                                         .thenCompose(etag ->
-                                                 store.getGroupProperties(namespace, group)
-                                                      .thenCompose(prop -> {
-                                                          return Futures.exceptionallyComposeExpecting(store.getSchemaVersion(namespace, group, schema),
-                                                                  e -> Exceptions.unwrap(e) instanceof StoreExceptions.DataNotFoundException,
-                                                                  () -> { // Schema doesnt exist. Validate and add it
-                                                                      return validateSchema(namespace, group, schema, prop.getCompatibility())
-                                                                              .thenCompose(valid -> {
-                                                                                  if (!valid) {
-                                                                                      throw new IncompatibleSchemaException(String.format("%s is incompatible", schema.getType()));
-                                                                                  }
-                                                                                  return store.addSchema(namespace, group, schema, prop, etag);
-                                                                              });
-                                                                  });
-                                                      })), executor)
+        return RETRY.runAsync(() ->
+                store.getGroupEtag(namespace, group)
+                     .thenCompose(etag ->
+                             store.getGroupProperties(namespace, group)
+                                  .thenCompose(prop -> {
+                                      return Futures.exceptionallyComposeExpecting(store.getSchemaVersion(namespace, group, schema, getFingerprint(schema)),
+                                              e -> Exceptions.unwrap(e) instanceof StoreExceptions.DataNotFoundException,
+                                              () -> { // Schema doesnt exist. Validate and add it
+                                                  return validateSchema(namespace, group, schema, prop.getCompatibility())
```

Comment:
Why do we validate with the normalized form rather than the original form?

Reply:
its the same. the validation involves checking for compatibility. the normalized and original form of schema will have the same answers for compatibility.

Reply:
In this case, we don't really need to produce the normalized form before this line, right? I'm trying to understand why we normalize early in this call and use in different calls.

Reply:
we normalize it before because we want to first check if the schema is already present or not --
`store.getSchemaVersion(namespace, group, schema, getFingerprint(schema))`

PS: i have renamed variables which hold the normalized schema to make it clearer where all they are used. 
Also, in the global pool, i am storing the first un-normalized schema binary submitted to the service instead of storing the normalized form there either.

Reply:
Please create the issue for the `get` call to return `null` rather than throw a data not found exception, and post the issue link here.

Reply:
#110